src/m365/teams/commands/chat/chat-message-send.ts

```diff
@@ -0,0 +1,244 @@
+import { GraphResponse } from '../../../base/GraphResponse';
+import GraphCommand from '../../../base/GraphCommand';
+import { AxiosRequestConfig } from 'axios';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  chatId?: string;
+  userEmails?: string;
+  chatName?: string;
+  message: string;
+}
+
+class TeamsChatMessageSendCommand extends GraphCommand {
+  public get name(): string {
+    return commands.CHAT_MESSAGE_SEND;
+  }
+
+  public get description(): string {
+    return 'Send a message to an existing or new chat conversation.';
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs, cb: () => void): Promise<void> {
+    try {
+      const chatId = args.options.chatId 
+        || args.options.userEmails && await this.ensureChatIdByUserEmails(args.options.userEmails)
+        || args.options.chatName && await this.getChatIdByName(args.options.chatName);
+      
+      await this.sendChatMessage(chatId as string, args.options);
+      cb();
+    } 
+    catch(error) {
+      this.handleRejectedODataJsonPromise(error, logger, cb);
+    }
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '--chatId [chatId]'
+      },
+      {
+        option: '--e, --userEmails [userEmails]'
+      },
+      {
+        option: '--chatName [chatName]'
+      },
+      {
+        option: '-m, --message <message>'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {    
+    if (!args.options.chatId && !args.options.userEmails && !args.options.chatName) {
+      return 'Specify chatId or userEmails or chatName, one is required.';
+    }
+
+    let nrOfMutuallyExclusiveOptionsInUse = 0;
+    if (args.options.chatId) { nrOfMutuallyExclusiveOptionsInUse++; }
+    if (args.options.userEmails) { nrOfMutuallyExclusiveOptionsInUse++; }
+    if (args.options.chatName) { nrOfMutuallyExclusiveOptionsInUse++; }
+
+    if (nrOfMutuallyExclusiveOptionsInUse > 1) {
+      return 'Specify either chatId or userEmails or chatName, but not multiple.';
+    }
+
+    if (!args.options.message) {
+      return 'Specify a message to send.';
+    }
+
+    if (args.options.chatId && !Utils.isValidTeamsChatId(args.options.chatId)) {
+      return `${args.options.chatId} is not a valid Teams ChatId.`;
+    }
+
+    if (args.options.userEmails) {
+      const userEmails = args.options.userEmails.toLowerCase().replace(/\s/g,"").split(",").filter(e => e && e !== "");
+      if (!userEmails || userEmails.length === 0 || userEmails.some(e => !Utils.isValidUserPrincipalName(e))) {
+        return `${args.options.userEmails} contains one or more invalid email addresses.`;
+      }
+    }
+
+    return true;
+  }
+  
+  private async ensureChatIdByUserEmails(userEmailsOption: string): Promise<string | undefined> {    
```

Comment:
Is there a valid case when this method will return `undefined` or is it always a `string`?